servicetalk-http-netty/src/test/java/io/servicetalk/http/netty/AbstractHttpServiceAsyncContextTest.java

```diff
@@ -190,33 +206,43 @@ private static List<Arguments> contextPreservedOverFilterBoundariesArguments() {
             for (boolean useImmediate : Arrays.asList(false, true)) {
                 for (boolean asyncService : Arrays.asList(false, true)) {
                     for (InitContextKeyPlace place : InitContextKeyPlace.values()) {
-                        params.add(Arguments.of(protocol, useImmediate, asyncService, place));
+                        for (ResponseType responseType : ResponseType.values()) {
+                            params.add(Arguments.of(protocol, useImmediate, asyncService, place, responseType));
+                        }
                     }
                 }
             }
         }
         return params;
     }
 
+    @RepeatedTest(10)
+    void repro() throws Exception {
+        //StreamingHttpServiceAsyncContextTest > contextPreservedOverFilterBoundaries(HttpProtocol, boolean, boolean, InitContextKeyPlace, ResponseType) > contextPreservedOverFilterBoundaries(HttpProtocol, boolean, boolean, InitContextKeyPlace, ResponseType) [162]:
+        // protocol=HTTP_2 useImmediate=false asyncService=false initContextKeyPlace=LIFECYCLE_OBSERVER_FILTER, responseType=CANCEL_ON_RESPONSE FAILED
+        contextPreservedOverFilterBoundaries(HttpProtocol.HTTP_2, false, false, InitContextKeyPlace.LIFECYCLE_OBSERVER, ResponseType.CANCEL_ON_RESPONSE);
```

Comment:
You added a lot more comprehensive coverage for different error/cancel scenarios, I didn't expect more than `ERROR_ON_RESPONSE_BODY` :) 
I didn't propose testing cancellation flow because it's hard due to racy events on client and server. If you run your new tests with coverage analysis, you will see that the `cancel()` inside new `afterFinally` is not invoked (at least locally). There is a chance it comes to the server in form of `onError`. But I see you are playing with `onRequestCancel()` assertions, so maybe on CI the behavior is different.